controllers/grafanacontactpoint_controller.go

```diff
@@ -207,6 +214,27 @@ func (r *GrafanaContactPointReconciler) reconcileWithInstance(ctx context.Contex
 	return nil
 }
 
+func (r *GrafanaContactPointReconciler) buildSettings(ctx context.Context, contactPoint *grafanav1beta1.GrafanaContactPoint) (models.JSON, error) {
+	marshaled, err := json.Marshal(contactPoint.Spec.Settings)
+	if err != nil {
+		return nil, fmt.Errorf("encoding existing settings as json: %w", err)
+	}
+	simpleContent, err := simplejson.NewJson(marshaled)
+	if err != nil {
+		return nil, fmt.Errorf("parsing marshaled json as simplejson")
+	}
+	for _, override := range contactPoint.Spec.ValuesFrom {
+		val, _, err := getReferencedValue(ctx, r.Client, contactPoint, override.ValueFrom)
+		if err != nil {
+			return nil, fmt.Errorf("getting referenced value: %w", err)
```

Comment:
if one of the overrides fail, we would also not apply any others. If we wanted that, we could log and `continue` here.

Reply:
I think it's better to fail early here to make sure we don't have a partially applied resource marked as success